RingPublishingGDPR/Classes/Private/Views/AppTrackingTransparencyView.swift

```diff
@@ -48,59 +93,61 @@ private extension AppTrackingTransparencyView {
-
-        var size = CGSize(width: image.size.width * scale, height: image.size.height * scale)
+        let textColor = UIColor(named: "ringPublishingGDPRLabel", in: Bundle.ringPublishingGDPRBundle, compatibleWith: nil)
 
-        let heightLimitRatio = logoImageView.bounds.height / size.height
-        size = CGSize(width: size.width * heightLimitRatio, height: size.height * heightLimitRatio)
+        let titleFontSize = titleTextView.font?.pointSize
+        let titleFont = uiConfig.font.withSize(titleFontSize ?? uiConfig.font.pointSize)
+        titleTextView.attributedText = attConfig?.attExplainationTitle?.convertfromHTML(using: titleFont,
+                                                                                        textColor: textColor)
 
-        return size
+        configureDescriptionText(attConfig?.attExplainationDescription, textColor: textColor, uiConfig: uiConfig)
     }
 
-    func configureTexts(with uiConfig: RingPublishingGDPRUIConfig) {
-        let textColor = UIColor(named: "ringPublishingGDPRLabel", in: Bundle.ringPublishingGDPRBundle, compatibleWith: nil)
+    func configureDescriptionText(_ text: String?,
+                                  textColor: UIColor?,
+                                  uiConfig: RingPublishingGDPRUIConfig?,
+                                  desiredFontSize: CGFloat? = nil) {
+        guard let fontSize = desiredFontSize ?? descriptionTextView.font?.pointSize,
+              let descriptionFont = uiConfig?.font.withSize(fontSize) else { return }
 
-        let titleFontSize = titleTextView.font?.pointSize
-        let titleFont = uiConfig.font.withSize(titleFontSize ?? uiConfig.font.pointSize)
-        titleTextView.attributedText = uiConfig.attOnboardingTitle?.convertfromHTML(using: titleFont,
-                                                                                    textColor: textColor)
+        descriptionTextView.attributedText = text?.convertfromHTML(using: descriptionFont, textColor: textColor)
+        descriptionTextView.layoutIfNeeded()
+
+        // Check if we have to shrink description font
+        let textViewSize = descriptionTextView.frame.size
+        let expectedSize = descriptionTextView.sizeThatFits(CGSize(width: textViewSize.width, height: CGFloat(MAXFLOAT)))
+
+        guard expectedSize.height > textViewSize.height else { return }
 
-        let descriptionFontSize = descriptionTextView.font?.pointSize
-        let descriptionFont = uiConfig.font.withSize(descriptionFontSize ?? uiConfig.font.pointSize)
-        descriptionTextView.attributedText = uiConfig.attOnboardingDescription?.convertfromHTML(using: descriptionFont,
-                                                                                                textColor: textColor)
+        configureDescriptionText(text, textColor: textColor, uiConfig: uiConfig, desiredFontSize: fontSize - 1)
```

Comment:
a może by tak jakoś to w pętli zrobić? że jak się nie mieści to dalej -1 i tak do 3 razy? ;)